packages/runtime/container-runtime/src/runWhileConnectedCoordinator.ts

```diff
@@ -59,38 +89,21 @@ export class RunWhileConnectedCoordinator {
      * The promise will also resolve if stopped either externally or by disconnect.
      * The return value indicates whether the start is successful or not.
      */
-    public async waitStart(): Promise<IStartedResult | INotStartedResult> {
-        if (!this.runtime.connected) {
-            if (this.everConnected) {
-                // We will not try to reconnect, so we are done running
-                return { started: false, message: "DisconnectedBeforeRun" };
-            }
+    public async waitStart() {
+        if (!this.runtime.connected && !this.everConnected) {
```

Comment:
We immediately return if we are connected or if we were connected sometime before but aren't connected anymore. How does a caller distinguish these two cases? Also, the promise returned can be resolved if cancelled.
Before this change, the return value could have been used. But now, does the caller have to check for connected after this call before proceeding further?

Reply:
The caller should never check for connected state - that is abstracted away by this call (cancellation token abstraction).
And yes, users of the system (caller and all the other layers that get cancellation token) need to check for cancellation frequently in various points to ensure that process moves to exit quickly on cancellation. waitStart() is not special here in any form - it's only special in a sense that this object needs waitStart() to be called to be functional.

Here is current usage :

```typescript
        const runCoordinator = new RunWhileConnectedCoordinator(this.runtime);
        await runCoordinator.waitStart();

        // Wait for either external signal to cancel, or loss of connectivity.
        const stopP = Promise.race([runCoordinator.waitCancelled, this.stopDeferred.promise]);
        void stopP.then((reason) => {
            this.logger.sendTelemetryEvent({
                eventName: "StoppingSummarizer",
                onBehalfOf,
                reason,
            });
        });

        if (runCoordinator.cancelled) {
            return;
        }
```

Reply:
Basically there is only one way of doing it, not two as before - waitStart() is no longer unique place that caller needs to worry about (other than need to call it).

Reply:
Okay. So, waitStart() is basically something that you need to call the first time after you create the coordinator. When it completes, the client is either connected or it was cancelled. The caller doesn't care about connected, it just needs to check that it is not cancelled before it proceeds.

Reply:
Ok, removing / hiding it allocher, by exposing static async create() method. It's much better now :)

Reply:
Okay cool. I was merely pointing out my understanding though :-)
Also nit: waitStart() should eventually become a private method.